src/sql/workbench/contrib/executionPlan/browser/executionPlanPropertiesViewBase.ts

```diff
@@ -258,10 +264,36 @@ export abstract class ExecutionPlanPropertiesViewBase extends Disposable impleme
 		this.resizeTable();
 	}
 
+	private repopulateTable() {
+		this._tableComponent.setData(this.flattenTableData(this._tableData, -1));
+		this.resizeTable();
+	}
+
 	public updateTableColumns(columns: Slick.Column<Slick.SlickData>[]) {
 		this._tableComponent.columns = columns;
 	}
 
+	public setPropertyRowsExpanded(expand: boolean): void {
+		this.setPropertyRowsExpandedHelper(this._tableComponent.getData().getItems(), expand);
+		this.repopulateTable();
```

Comment:
You could probably further optimize this by not needing to re-write the table data (after all - the treegrid already has all the data). You just need to update the properties on the data in the treegrid itself and force it to re-render.